gpt_engineer/chat_to_files.py

```diff
@@ -40,3 +43,83 @@ def to_files(chat, workspace):
     files = parse_chat(chat)
     for file_name, file_content in files:
         workspace[file_name] = file_content
+
+
+def overwrite_files(chat, dbs, replace_files):
+    """
+    Replace the AI files to the older local files.
+    """
+    dbs.workspace["all_output.txt"] = chat
+
+    files = parse_chat(chat)
+    for file_name, file_content in files:
+        # Verify if the file created by the AI agent was in the input list
+        if file_name in replace_files:
+            # If the AI created a file from our input list, we replace it.
+            with open(replace_files[file_name], "w") as text_file:
+                text_file.write(file_content)
+        else:
+            # If the AI create a new file I don't know where to put it yet
+            # maybe we can think in a smarter solution for this in the future
+            # like asking the AI where to put it.
+            #
+            # by now, just add this to the workspace inside .gpteng folder
+            print(
+                f"Could not find file path for '{file_name}', creating file in workspace"
+            )
+            dbs.workspace[file_name] = file_content
```

Comment:
Hm I think workspace should be == cwd no?

Reply:
I’d be fine with not checking and just using chat_to_files

the improvement might be to refactor into new files right?

Reply:
The problem I see is the creation of "all_output.txt" in the root of the folder. 

Also, if I only use "to_files" function I'll be limited to the files from the workspace. With the function "ovewrite_files" we can change files in any folder. 

By now I'll use the workspace as cwd, but I'll continue using the overwrite function.

Reply:
We can just skip writing all_output.txt. It should be in memory/logs regardless!

Reply:
Would be so nice to simplify this function!

Less code is always better.

I.e.: We set workspace to the current directory, and get rid of the "open(filename)" boilerplate by always doing `dbs.workspace[name] = content`